easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -2234,6 +2234,22 @@ def resolve_template(value, tmpl_dict, expect_resolved=True):
     return value
 
 
+def _copy_ec_dict(easyconfig):
+    """Copy an easyconfig dict as (initially) parsed"""
+    ec = easyconfig.pop('ec')  # Don't copy this via deepcopy
+    try:
+        new_easyconfig = copy.deepcopy(easyconfig)  # Copy the rest of the dict
```

Comment:
Can you clarify why this is in a `try`?

What if this fails? Then `new_easyconfig` will be undefined, and we'll get a nasty crash, no?